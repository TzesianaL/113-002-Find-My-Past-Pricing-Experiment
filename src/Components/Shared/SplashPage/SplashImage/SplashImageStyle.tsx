import styled from 'styled-components'
/**
 * Styled component is used to decorate SplashImage component.
 * Props are passed from ThemeProvider component.
 */
export const SplashImageStyle = styled.div`
  display: grid;
  justify-self: center;
  align-self: center;
  width: ${(props) => props.theme.splashImage?.imageSize || 100}%;
  img {
    width: 100%;
  }
`
