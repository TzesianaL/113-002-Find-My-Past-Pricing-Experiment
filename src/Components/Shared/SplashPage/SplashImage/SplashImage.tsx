import React, { FC } from 'react'
import { SplashImageStyle } from './SplashImageStyle'

export interface SplashImgProps {
  /**
   * Image path for image that displays on the splashscreen component.
   */
  imageSrc: string
}

/**
 * SplashImg component is used to display any images included in the splashScreen
 * SplashImgStyle component is used to style the component
 */
const SplashImg: FC<SplashImgProps> = ({ imageSrc }) => {
  return (
    <SplashImageStyle>
      <img src={imageSrc} alt="splashimg" />
    </SplashImageStyle>
  )
}

export default SplashImg
