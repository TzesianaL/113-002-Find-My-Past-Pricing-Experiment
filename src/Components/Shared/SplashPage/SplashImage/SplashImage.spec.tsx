import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import SplashImage from './SplashImage'

describe('<SplashImage/>', () => {
  it('should renderer correctly SplashText', () => {
    const tree = renderer
      .create(<SplashImage imageSrc="test source" />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
