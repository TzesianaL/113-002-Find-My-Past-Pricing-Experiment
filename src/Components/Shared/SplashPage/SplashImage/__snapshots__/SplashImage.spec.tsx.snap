// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashImage/> should renderer correctly SplashText 1`] = `
.c0 {
  display: grid;
  justify-self: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 100%;
}

.c0 img {
  width: 100%;
}

<div
  className="c0"
>
  <img
    alt="splashimg"
    src="test source"
  />
</div>
`;
